### ISSUE TITLE
Massive pixel shifting

### ISSUE THREAD

**Issue by buggedcom** [amcharts/amcharts5#586]
**Question**

Our Amcharts setup uses a massive react component chain that is not easily separated into a codepen. So therefore I ask, has anyone experienced, what I have coined as "disco mode", with Amcharts?

If so, or even if not, any ideas as to what would cause this?

My initial thoughts were that this would be down to the width and height settings, changing dynamically, however, I have checked that this is not the case and the size being passed down to both Amcharts and the parent containers are of constant width and height. So I'm somewhat of a loss as to what to try next.

https://user-images.githubusercontent.com/2810/187206858-0da08434-caa4-412c-8339-72798a7df0d3.mov
https://user-images.githubusercontent.com/2810/187206887-b4bd144d-0a21-47dd-8d5e-53d3f01af427.mov

**Environment (if applicable)**

This is with AmCharts 5.2.19 and any browser.

**Comment by martynasma** [comment]
I think we fixed an issue like that in 5.2.18.

It was related to Y axis having adapters on their labels.

Is that the case with your setup?

Also, can you doublecheck that you are absolutely using 5.2.19?

If there are no adapters in use, or you are running latest version, than it's a completely different issue. I'm afraid we'll need some way to reproduce it.

**Comment by buggedcom** [comment]
Yes, I just spotted the [related issue](https://github.com/amcharts/amcharts5/issues/564) as well. Removing the adapters (or at least making them return an empty string) fixes the issue.

I will double check the issue that we have 5.2.19 running correctly and then if so try to re-produce a workable codepen for you now that I have isolated the issue.

**Comment by buggedcom** [comment]
Ok, I've found the root cause. It is because within the return text of the formatted we are returning a custom styling for the label.

For example:

```js
renderer.labels.template.adapters.add('text', (text, target) => {
    return '[fontFamily: "Space Mono"]' + text + ' kWh[/]';
});
```

Removing the styling and the problem does no longer exist.

```js
renderer.labels.template.adapters.add('text', (text, target) => {
    return '[]' + text + ' kWh[/]';
});
```

However, what is more curious is that this code previously worked fine when adding series data to the chart after it had already been created, but when the chart is created with series from the start it has started doing this. So it is likely some kind of setup sequencing issue related to the fonts not being ready to correctly size them.

I will attempt to setup and codepen now

**Comment by buggedcom** [comment]
Repeatable pen

https://codepen.io/buggedcom/pen/bGvPXvq?editors=0010

**Comment by buggedcom** [comment]
Note, however that it is only on first load of the page. If you edit the pen, which rebuilds the page, the chart works fine.

**Comment by buggedcom** [comment]
Interestingly leaving the formatting in place but using a non-mono spaced font such as `"Rubik"` does not have this issue. So it could be something to do with mono-space fonts.

**Comment by martynasma** [comment]
Thanks for the CodePen!

It most certainly is related to loadable fonts.

We'll see what we can do.

**Comment by martynasma** [comment]
Fixed in [5.2.20](https://github.com/amcharts/amcharts5/releases/tag/5.2.20).

## [5.2.20] - 2022-09-05

### Added
- All indicator series will now have `"indicator"` theme tag set. Detect indicator series via `series.hasTag("indicator")`.
- `autoZoom` added to `ValueAxis` (default: `true`). If set to `false`, the axis won't be auto-zoomed to a selection (this works only if the other axis is a `DateAxis` or a `CategoryAxis`). This setting will be ignored if both X and Y axes are a `ValueAxis`.
- `StandardDeviation` indicator added.

### Changed
- Series edit modal on a `StockChart` would show incorrect controls if the main chart series was a `LineSeries`.

### Fixed
- Initial sizing of HTML tooltips fixed (again).
- Exporting a chart in a `<div>` with `height: auto` was not working properly.
- If a point in a `LineSeries` was way out of bounds (million times further than the bounds of plot area), the line could disappear.
- In some cases, if chart was using loadable fonts, it could go into non-stop flickering mode.
- `StockChart` with drawing mode disabled could start drawing new objects if clicked on points of an old drawing.
- `PointedRectangle`, mostly used by a `Tooltip` was drawn incorrectly if its stem size was bigger than height. 
- Removing a `QuadrantLineSeries` drawing from a `StockChart` resulted an error.
- It was impossible to delete `Fibonacci` or `FibonacciTimezone` drawings one by one after switching to another drawing tool.

[Full change log](https://github.com/amcharts/amcharts5/blob/master/packages/shared/CHANGELOG.md).

[Download options](https://www.amcharts.com/download/).

Make sure you clear your browser cache after upgrading. And feel free to contact us again if you are still experiencing this issue.

**Comment by buggedcom** [comment]
@martynasma This appears to be also happening vertically now. I'm assuming it's related and I don't need to produce a pen. however if not, I will produce a pen.


https://user-images.githubusercontent.com/2810/210254847-9b7d6cb5-8ce6-40ed-bd39-820bc111b3a5.mov
